### PR TITLE
more verbose multi-line display(c) for Char

### PR DESCRIPTION
**Pull request by stevengj**
`show('$')` prints a terse representation like `'$'`.   I've been thinking it would be nice to have a more informative display in the REPL (and in other things like IJulia that use the 3-argument `show` via `display`).  This PR does so, giving output like:
```
julia> '$'
'$': ASCII/Unicode U+0024 (category Sc: Symbol, currency)

julia> '⊗'
'⊗': Unicode U+2297 (category Sm: Symbol, math)
```

### CODE REVIEW

File: base/char.jl [hunk 1]
```diff
@@ -77,3 +77,11 @@ function show(io::IO, c::Char)
     end
     return
 end
+
+using UTF8proc: category_abbrev, category_string
+function show(io::IO, ::MIME"text/plain", c::Char)
+    show(io, c)
+    u = UInt32(c)
+    print(io, ": ", isascii(c) ? "ASCII/" : "", "Unicode U+", hex(u, u > 0xffff ? 8 : 4))
```
Comment (JeffBezanson) [inline base/char.jl]: Maybe use 6 instead of 8?

**Comment by quinnj** [comment]
What is `category Sc` vs `category Sm`? Should it maybe just be `category: Symbol, currency`? What are the non-Symbol characters? control characters?

**Comment by stevengj** [comment]
Both the two-letter abbreviations and the expanded names for the [Unicode character categories](http://www.fileformat.info/info/unicode/category/index.htm) are standard, so it seemed useful to give them both.

File: base/char.jl (hunk 1 above)
Comment (stevengj) [inline base/char.jl]: ok

**Comment by TotalVerb** [comment]
Would the Unicode description, like

```
'$': Unicode U+0024 Dollar sign (category Sc ...)
```

be possible/desirable to add?

**Comment by stevengj** [comment]
@TotalVerb, not without storing the Unicode character-name database somewhere; it's a couple-hundred kB uncompressed IIRC (see also #19725 and related issues), so I'm not sure this is a great thing for us to do.

**Comment by stevengj** [comment]
One option that occurred to me is to have a `show(io, "text/html", c::Char)` method that links to the complete description of the codepoint on e.g. www.fileformat.info, for example [U+0302](http://www.fileformat.info/info/unicode/char/0302/index.htm), but that doesn't help in the REPL.

**Comment by stevengj** [comment]
It also occurred to me to output the tab-completion info, which @ScottPJones also suggested on the mailing list, but this seemed too REPL-specific.   It seemed better to just have this in the REPL help, e.g. if you type `?⊗` it should tell you that you can type `⊗` with `\otimes<tab>`, in addition to any other help.

**Comment by stevengj** [comment]
I'm not understanding the Travis failure here...

**Comment by stevengj** [comment]
Should be good to merge; the i686 travis failure seems to be an endemic unrelated problem at the moment.

**Comment by StefanKarpinski** [comment]
I like it. Should we make sure that `repr(c)` is still just `'x'`?

**Comment by stevengj** [comment]
I added a test for `repr`.   (`repr` uses the 2-argument `show`, which hasn't been changed.)

**Comment by stevengj** [comment]
Should be good to squash/merge; the Travis failure is the unrelated x86 problem we are having nowadays.

**Pull request merged**